src/dtx/dtx_common.c

```diff
@@ -519,7 +519,7 @@ dtx_leader_end(struct dtx_leader_handle *dlh, struct ds_cont_hdl *cont_hdl,
 	/* NB: even the local request failure, dth_ent == NULL, we
 	 * should still wait for remote object to finish the request.
 	 */
-	rc = dtx_leader_wait(dlh, &dces, &dces_cnt);
+	rc = dtx_leader_wait(dlh, &dces, result >= 0 ? &dces_cnt : NULL);
```

Comment:
Ah, I see. Thanks for the explanation. Just curious, when result could > 0?

Reply:
Sorry, I cannot remember why the original implementation (before your re-org of the code) was ">= 0", but not "== 0". It something like for the case of vos or dtx module to return positive value. Anyway, it is consistent with the dtx_{leader}_end for the "result" handling.